tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Framework.cs

```diff
@@ -282,6 +284,24 @@ internal void BlockAndReport(Dictionary<string, object> args, bool lastWafCall =
         }
     }
 
+    internal void BlockAndReport(IResult? result)
```

Comment:
How is this different to the `ReportAndBlock()` bellow, and how do you know which one to choose? 😅

Reply:
yes I must say it's a bit confusing, `ReportAndBlock` is used by rasp, as the blocking middleware is not always here to catch the exception, so to make sure it's reported it's reporting before blocking it.
But I think the preferred method should always be `BlockAndReport` as we're reporting the actual http status code and other stuff, we want to make sure we report what currently is and not what _will_ be theoretically after blocking...